### PR TITLE
[feat] create full size button for onboarding and auth

### PR DESCRIPTION
**Pull request by rahihazra**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

## :tanabata_tree: Description
- Creates a full size button to be used for the onboarding and auth pages

### :palm_tree: What's new in this PR
[//]: # "REQUIRED - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- there is a customized button component (that will be changed in the future due to design)
- since we had no place to put the button, i created a test page to display only the component

### :evergreen_tree: Screenshots
[//]: # "REQUIRED for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! If you are making changes here, please CC: @davidqing6432 at the bottom."
Screen Shot 2023-10-26 at 1 30 02 PM

## :deciduous_tree: How to review
[//]: # 'REQUIRED - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
- order in which to review: src/components/BigButton.tsx and then src/app/button-test/page.tsx
- we don't have a specific place to implement this button, so changes will need to be made

## :seedling: Next steps 
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### :information_source: Online sources
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### :potted_plant: Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
CC: @pragyakallanagoudar

### CODE REVIEW

File: src/components/BigButton.tsx
Comment (jinkang-0) [inline src/components/BigButton.tsx]: I think the convention for styled component props is to have a `$` prefix attached to them, so like `$color`, which is just to differentiate them from React components I think? Not too sure what conventions we're following though.

File: src/components/BigButton.tsx
Comment (jinkang-0) [inline src/components/BigButton.tsx]: We don't need to add an `onChange` prop here, since the `BigButton` builds off of a regular HTML button (signified by `styled.button`), it already has `onChange` as a possible prop, so we don't need to (and can't, in styled components) handle that!

In the future, it's useful to keep in mind that props to a styled component is meant to be help decide how to style. In the case of `onChange`, it won't help us decide how to style this button (i.e. what color, what padding, etc etc)

File: src/components/BigButton.tsx
Comment (jinkang-0) [inline src/components/BigButton.tsx]: Ideally it's good practice to use `rem` units instead of `px`! By default, most browsers have it set such that `1rem` = `16px`, so we can do unit conversion to adjust that. The reason why is so things scale nicely with `font-size`.

File: src/components/FullSizeButton.tsx
Comment (jinkang-0) [inline src/components/FullSizeButton.tsx]: Not sure what this file is used for...? Can we remove this file?

File: src/app/button-test/page.tsx
Comment (jinkang-0) [inline src/app/button-test/page.tsx]: Nice! I think we should remove testing pages before merging them in ideally though, but maybe you can comment this out and move it into `BigButton.tsx` to show how to use it?

File: src/components/BigButton.tsx
Comment (jinkang-0) [inline src/components/BigButton.tsx]: Flagging this here as well for `rem` unit conversion! `border-radius` is one of those things that can get finicky, but I think we do want it to scale with `font-size`.

**Review by jinkang-0 — COMMENTED** [review]
Looks good so far! I left some comments for improvements, but it should be good to merge after they're addressed!

File: src/components/BigButton.tsx
Comment (pragyakallanagoudar) [reply src/components/BigButton.tsx]: Let's change to this! Thanks Jinkang

File: src/components/BigButton.tsx
Comment (pragyakallanagoudar) [reply src/components/BigButton.tsx]: Agree! Also, since we're only passing in the one color prop, I think we can go ahead and get rid of this interface. Instead, your styled component would look something like:

```
const BigButton = styled.button<{ $color?: string }>`
// ...etc.
```

File: src/app/button-test/page.tsx
Comment (pragyakallanagoudar) [reply src/app/button-test/page.tsx]: +1! let's keep this here for right now because we're not ready to merge yet. once we are, we can do what Jinkang recommended

Comment (pragyakallanagoudar) [reply src/app/button-test/page.tsx]: Also, let's pass in some props to the button so we can test it:

```
<BigButton color={COLORS.abaBlue} onChange={() => console.log('button clicked')}>
      Big Button
</BigButton>
```

File: src/components/BigButton.tsx
Comment (pragyakallanagoudar) [reply src/components/BigButton.tsx]: https://nekocalc.com/px-to-rem-converter

**Review by jinkang-0 — APPROVED** [review]
lgtm!

**Review by pragyakallanagoudar — APPROVED** [review]
Looks great! For reference, we commented out a bunch of things to prevent breaking from schema changes, and also commented out the button test screen and moved it to the Button file.

**Pull request merged**
